python/ray/data/dataset.py

```diff
@@ -2311,6 +2321,52 @@ def train_test_split(
                 )
             return ds.split_at_indices([ds_length - test_size])
 
+    def _stratified_split(
+        self, ds: "Dataset", test_size: Union[int, float], stratify: str
+    ) -> Tuple["MaterializedDataset", "MaterializedDataset"]:
+        """Perform stratified train-test split on the dataset.
+        
+        Args:
+            ds: The dataset to split.
+            test_size: Test size as int or float.
+            stratify: Column name to use for stratified sampling.
+            
+        Returns:
+            Train and test subsets as two MaterializedDatasets.
+        """
+        # Normalize test_size to float (only materialize if needed)
+        if isinstance(test_size, int):
```

Comment:
can you unify the validation code with the original train_test_split to avoid duplication?